### DISCUSSION TITLE
Initializing Config with wallet connected elsewhere (e.g. nested WagmiConfigs)

### DISCUSSION THREAD

**Discussion by kasparkallas** [wevm/wagmi#2307] [General]
I'm developing a widget that interacts with smart contracts. The widget can be pulled in as an NPM package and included in the dApp as a React component. I'd like the widget to use wagmi's React hooks but not necessarily do the wallet connection management.

An example:
The widget could be included in a dApp that's already using wagmi with connectors defined and the wallet connected. The widget should take Wallet Client as an input and have the internal `WagmiConfig` associated with the input.

Before v1, it seemed that I could use the contract write hooks by passing in ether's Signer. After v1, ~~the transaction signing part seems to be more implicit, i.e. you can't pass in the signing part~~. EDIT: Oh wait, you can... (`walletClient`)

Can I get some quick guidance on the matter? 

What I'm thinking:
* wagmi is meant as a top-level thing and I shouldn't consider nesting instances of it. Use viem with my own hooks instead.
* Implemented a custom connector for the nested instance that would wrap the Wallet Client passed in.

**Comment by dvartic** [comment]
@kasparkallas Were you able to initialize config with a signer? I understand that you can use hooks passing a custom signer every time, but how about initializing the global config with it?
